libtrap/src/ifc_tls.c

```diff
@@ -1827,76 +1934,85 @@ int create_tls_sender_ifc(trap_ctx_priv_t *ctx, const char *params, trap_output_
       result = TRAP_E_BADPARAMS;
       goto failsafe_cleanup;
    }
-   if (param_iterator != NULL) {
-      /* still having something to parse... Rotate all parameters, the first one is probably max_clients*/
-      max_clients = keyfile;
-      keyfile = certfile;
-      certfile = cafile;
-      param_iterator = trap_get_param_by_delimiter(param_iterator, &cafile, TRAP_IFC_PARAM_DELIMITER);
-   }
-   if (max_clients == NULL) {
-      /* 2nd parameter became optional, set default value when missing */
-      max_num_client = TRAP_IFC_DEFAULT_MAX_CLIENTS;
-   } else {
-      if (sscanf(max_clients, "%u", &max_num_client) != 1) {
-         VERBOSE(CL_ERROR, "Optional max client number given, but it is probably in wrong format.");
-         max_num_client = TRAP_IFC_DEFAULT_MAX_CLIENTS;
+
+   /* Optional params */
+   while (param_iterator != NULL) {
+      param_iterator = trap_get_param_by_delimiter(param_iterator, &param_str, TRAP_IFC_PARAM_DELIMITER);
+      if (param_str == NULL)
+         continue;
+      if (strncmp(param_str, "buffer_count=x", 13) == 0) {
+         if (sscanf(param_str+13, "%u", &buffer_count) != 1) {
```

Comment:
Coding style: `param + 13`
+ it would be better to have a named constant / CPP macro instead of 13

Reply:
Fixed, constants are defined in ifc_socket_common.h